### DISCUSSION TITLE
Loading in the level.dat file from Anvil Worlds

### DISCUSSION THREAD

**Discussion by GreatWyrm** [Minestom/Minestom#959] [Help] [answered]
The NBT and Tag system has changed quite a bit over the past few months, and currently I'm struggling to figure out how to read in the World Spawnpoint.
From the [wiki](https://wiki.vg/Map_Format), it's under the Data compound, so I read it in like so: `NBT worldLevelDat = world.getTag(Tag.NBT("Data"));`

However, I'm not sure where to go from there. The methods on the NBT class itself are unhelpful, and I'm not sure how to use the other classes (NBTCompound and MutableNBTCompound) to get the integers I am looking for.

If anyone knows how to do that, I would appreciate it!

**Comment by kiip1** [comment, marked as answer]
You can read integers very easily from the compound like this:
```java
NBTCompound compound = (NBTCompound) instance.getTag(Tag.NBT("Data"));
Integer spawnX = compound.getInt("spawnX");
Integer spawnY = compound.getInt("spawnY");
Integer spawnZ = compound.getInt("spawnZ");

if (spawnX != null && spawnY != null && spawnZ != null) {
    // Use them here...
}
```
